src/ruby/lib/grpc/generic/rpc_server.rb

```diff
@@ -354,6 +354,59 @@ def run
 
     alias_method :run_till_terminated, :run
 
+    # runs the server with signal handlers
+    # @param signals
+    #     List of String, Integer or both representing signals that the user
+    #     would like to send to the server for graceful shutdown
+    # @param wait_interval (optional)
+    #     Integer seconds that user would like stop_server_thread to poll
+    #     stop_server
+    def run_till_terminated_or_interrupted(signals, wait_interval = 60)
+      valid_signals = Signal.list
+
+      stop_server = false
+      stop_server_cv = ConditionVariable.new
+      stop_server_mu = Mutex.new
+      stop_server_thread = Thread.new do
+        loop do
+          break if stop_server
```

Comment:
Realizing one edge case in this API: this probably <i>should</i> also handle the case where the server was one stopped the server with by calling the `stop` method on it, rather than necessarily interrupting it with a signal. For example, if one used this API to start their server, they should be able to interrupt it and/or call `stop` on it, and it should then begin shut down.
One idea for how that could be done is we could change `stop_server_cv`, `stop_server_mu`, and `stop_server` from local vars of this function to instance-state for the whole `RpcServer`. Then, `RpcServer#stop` could then broadcast the condition variable and set `stop_server = true`. Currently, I think it will just hang if `stop` is used, until a signal is received.